### DISCUSSION TITLE
Question about ODataSettings  and ODataQuerySettings.EnableConstantParameterization

### DISCUSSION THREAD

**Discussion by amorelIM** [AutoMapper/AutoMapper.Extensions.OData#212] [Q&A] [answered]
Hi,

I was wondering if there is any plan to support the EnableConstantParameterization parameter in the ODataSettings object? We
 use your package, and in some use cases, we would like to set the ODataQuerySettings.EnableConstantParameterization to false to get better performance with some queries. 

For example, when we use AutoMapper.Extensions.OData EnableConstantParameterization is always true, so parameters are used in queries and some characters are escaped.  But when looking at the execution plan of SQL Server 2019, we found that it cannot parallelize queries where characters are escaped in a WHERE clause. When EnableConstantParameterization is set to false, it takes 0.3 second to retrieve some data, compared to 4.1 seconds when the same parameter is set to true.

**Comment by BlaiseD** [comment, marked as answer]
No plans but PRs are welcome.  You'll want to include [failing tests](https://github.com/AutoMapper/AutoMapper.Extensions.OData/blob/master/AutoMapper.OData.EFCore.Tests/GetQueryTests.cs) which pass with you changes.

**Reply by alexmo16** [reply]
Here is the PR https://github.com/AutoMapper/AutoMapper.Extensions.OData/pull/215 !
